File: week01/005.pow.c
// Calculating compound interest
#include <stdio.h>
#include <math.h>

int main(void) {
    double amount;
    double principal = 1000.0;
    double rate = 0.05;      // annual interest rate
    unsigned int year;

    // output table header
    printf("%4s%21s\n", "Year", "Amount on deposit");

    for( year=1; year <= 10; year++) {
        amount = principal * pow(1.0 + rate, year);

        printf("%4u%21.2f\n", year, amount);
    }

}

File: week01/005.pow.rs
// Calculating compound interest
// https://doc.rust-lang.org/nightly/std/primitive.f32.html#method.powi
// use std::io; // not used
use std::f32; //powi, powf ...

fn main() {
    let mut amount: f32;
    let principal: f32 = 1000.0;
    let rate: f32 = 0.05;

    println!("{:4} {:21}", "Year", "Amount on deposit");

    for year in 1..11 {
        amount = principal * (1.0 + rate).powi(year);

        println!("{:4} {:21.2}", year, amount);
    }

}
